src/lightning/fabric/CHANGELOG.md

```diff
@@ -118,6 +118,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed computing the next version folder in `CSVLogger` ([#17139](https://github.com/Lightning-AI/lightning/pull/17139))
 
 
+- Fixed validation of parameters of `plugins.precision.MixedPrecision` ([#17687](https://github.com/Lightning-AI/lightning/pull/17687))
```

Comment:
I'm not sure this classifies as a bug fix. The type hints were correct. I would say this is an "addition" of extra validation.

Reply:
Hm, as I outlined in the ticket, I would consider it a bug for the following reasons.
- If a wrong configuration is passed, the plugin silently disrupts the training. In detail, it changes the precision but does not adjust the gradients accordingly. IMO the whole purpose of this plugin is to avoid this kind of inconsistent behavior.
- The valid configuration value changed from versions <2.0 to now. Thus upgrading makes it likely to inadvertently pass a wrong configuration value.

I get your point with type hints. However, that only triggers an alert if the value is passed as a literal - in many cases if it is read from a config file. Thus there will be no warning at all.